subsys/bluetooth/host/conn.c

```diff
@@ -690,7 +740,7 @@ static int send_buf(struct bt_conn *conn, struct net_buf *buf)
 	LOG_DBG("conn %p buf %p len %u", conn, buf, buf->len);
 
 	/* Send directly if the packet fits the ACL MTU */
-	if (buf->len <= conn_mtu(conn) && !tx_data(buf)->is_cont) {
+	if (fits_single_ctlr_buf(buf, conn) && !tx_data(buf)->is_cont) {
```

Comment:
should `while (buf->len > conn_mtu(conn)) {` below in line 758 be fixed?

Reply:
actually I think it should be fine: it's only the first fragment that has the data header (that doesn't count against the conn_mtu). 
So as long as the controller doesn't set its TX size to 0 (ie the mtu >= data hdr size) then this will work as expected.

But you're right that something is awry: the length of the data copied in `send_frag` doesn't take the header into account, now I fixed it too.